From 105ebfb49a381d9ecfc7f83b6a31aa2e851e4992 Mon Sep 17 00:00:00 2001
From: Johannes Passing <jpassing@hotmail.com>
Date: Thu, 4 Mar 2021 16:06:31 +0100
Subject: [PATCH 1/2] Handle OSC sequences with empty text parameters

Previously, the parser only handled sequences with non-empty text
parameters and failed to handle sequences like `\e]2;\a` or `\e]0;\a`.

This caused some programs to hang because they did not receive
the expected feedback.
---
 VtNetCore.Unit.Tests/libvtermStateTerminalProperties.cs | 6 ++++++
 VtNetCore/XTermParser/XTermSequenceReader.cs            | 1 +
 2 files changed, 7 insertions(+)

diff --git a/VtNetCore.Unit.Tests/libvtermStateTerminalProperties.cs b/VtNetCore.Unit.Tests/libvtermStateTerminalProperties.cs
index 22f9ef3..67829bb 100644
--- a/VtNetCore.Unit.Tests/libvtermStateTerminalProperties.cs
+++ b/VtNetCore.Unit.Tests/libvtermStateTerminalProperties.cs
@@ -125,6 +125,12 @@
                 windowTitle = args.Text;
             };
 
+            // !Title
+            // PUSH "\e]2;\a"
+            //   settermprop 4 ""
+            Push(d, "".ChangeWindowTitle(""));
+            Assert.Equal("", windowTitle);
+
             // !Title
             // PUSH "\e]2;Here is my title\a"
             //   settermprop 4 "Here is my title"
diff --git a/VtNetCore/XTermParser/XTermSequenceReader.cs b/VtNetCore/XTermParser/XTermSequenceReader.cs
index 1df90e6..1e55f21 100644
--- a/VtNetCore/XTermParser/XTermSequenceReader.cs
+++ b/VtNetCore/XTermParser/XTermSequenceReader.cs
@@ -169,6 +169,7 @@
 
                         Parameters.Add(currentParameter);
                         currentParameter = -1;
+                        readingCommand = true;
                     }
                     else if (char.IsDigit(next))
                     {
-- 
2.17.1.windows.2

